subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/maven/RealisedMavenModuleResolveMetadata.java

```diff
@@ -101,15 +103,30 @@ public static RealisedMavenModuleResolveMetadata transform(DefaultMavenModuleRes
             ImmutableAttributes variantAttributes = variantMetadataRules.applyVariantAttributeRules(variant, metadata.getAttributes());
             CapabilitiesMetadata capabilitiesMetadata = variantMetadataRules.applyCapabilitiesRules(variant, ImmutableCapabilities.EMPTY);
 
+            ImmutableList.Builder<MavenDependencyDescriptor> descriptors = findDependencyDescriptors(metadata, configurationName);
             RealisedConfigurationMetadata realisedConfiguration = createConfiguration(variantMetadataRules, metadata.getId(), configurationName, configuration.isTransitive(), configuration.isVisible(),
-                LazyToRealisedModuleComponentResolveMetadataHelper.constructHierarchy(configuration, configurationDefinitions), metadata.getDependencies(),
+                LazyToRealisedModuleComponentResolveMetadataHelper.constructHierarchy(configuration, configurationDefinitions), descriptors.build(),
                 variantAttributes, ImmutableCapabilities.of(capabilitiesMetadata.getCapabilities()));
             configurations.put(configurationName, realisedConfiguration);
         }
         RealisedMavenModuleResolveMetadata realisedMavenModuleResolveMetadata = new RealisedMavenModuleResolveMetadata(metadata, variants, derivedVariants, configurations);
         return realisedMavenModuleResolveMetadata;
     }
 
+    private static ImmutableList.Builder<MavenDependencyDescriptor> findDependencyDescriptors(DefaultMavenModuleResolveMetadata metadata, String configurationName) {
+        List<? extends DependencyMetadata> dependencies = metadata.getConfiguration(configurationName).getDependencies();
+        ImmutableList.Builder<MavenDependencyDescriptor> descriptors = ImmutableList.builder();
+        for (DependencyMetadata dependency : dependencies) {
+            if (dependency instanceof ConfigurationBoundExternalDependencyMetadata) {
+                ExternalDependencyDescriptor dependencyDescriptor = ((ConfigurationBoundExternalDependencyMetadata) dependency).getDependencyDescriptor();
+                if (dependencyDescriptor instanceof MavenDependencyDescriptor) {
+                    descriptors.add((MavenDependencyDescriptor) dependencyDescriptor);
+                }
+            }
+        }
+        return descriptors;
```

Comment:
This could return the built list instead of the builder.

Reply:
Indeed